### PR TITLE
fix: handle non terminal sessions based on command

### LINKED ISSUE #602: CLI commands break in non terminal environments
### Issue
Any bubbletea based command will fail on non terminal environments. We can fix this by detecting non terminal and switching to non interactive mode.

### Steps

Use a tool like concurrently to test the bug:
```
    "start": "concurrently --kill-others-on-fail \"nitric start\" \"wait-port localhost:50051 && npm run dev:functions\"",
```

### Expected
Switch to non interactive mode

### PR DESCRIPTION
**Pull request by davemooreuws**
Will either switch to non-interactive mode or error depending on the command. Open to changing this logic if it makes sense.

fixes #602

### CODE REVIEW

File: pkg/utils/terminal.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"os"
+
+	"golang.org/x/term"
+)
+
+func IsTerminal() bool {
+	if term.IsTerminal(int(os.Stdin.Fd())) && term.IsTerminal(int(os.Stdout.Fd())) {
+		return true
+	}
```
Comment (tjholm) [inline pkg/utils/terminal.go]: To make sure this is compatible with bubbletea should we use `termenv` if methods exist for detecting this?

Comment (davemooreuws) [reply pkg/utils/terminal.go]: switched over method

**Review by HomelessDinosaur — DISMISSED** [review]

File: cmd/new.go [hunk 2]
```diff
@@ -45,6 +48,10 @@ nitric new hello-world "official/TypeScript - Starter" `,
 			templateName = args[1]
 		}
 
+		if !utils.IsTerminal() && (templateName == "" || projectName == "") {
+			return fmt.Errorf(`non-terminal detected, for a non-interactive environment use the arguments: nitric new hello-world "official/TypeScript - Starter"`)
```
Comment (jyecusch) [inline cmd/new.go]:
```suggestion
			return fmt.Errorf(`non-interactive environment detected, please provide all mandatory arguments e.g. nitric new hello-world "official/TypeScript - Starter"`)
```

File: cmd/run.go [hunk 3]
```diff
@@ -42,6 +44,12 @@ var runCmd = &cobra.Command{
 		log.SetOutput(output.NewPtermWriter(pterm.Debug))
 		log.SetFlags(0)
 
+		if !utils.IsTerminal() && !output.CI {
+			fmt.Println("")
+			pterm.Warning.Println("Non-terminal detected, switching to non-interactive mode")
```
Comment (jyecusch) [inline cmd/run.go]:
```suggestion
			pterm.Warning.Println("non-interactive environment detected, switching to non-interactive mode")
```

File: cmd/stack.go [hunk 4]
```diff
@@ -84,6 +84,10 @@ var newStackCmd = &cobra.Command{
 	Short: "Create a new Nitric stack",
 	Long:  `Creates a new Nitric stack.`,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		if !utils.IsTerminal() {
+			return fmt.Errorf("non-terminal detected, the stack new command does not support non-interactive mode")
```
Comment (jyecusch) [inline cmd/stack.go]:
```suggestion
			return fmt.Errorf("the stack new command does not support non-interactive environments")
```

File: cmd/stack.go [hunk 5]
```diff
@@ -135,6 +139,12 @@ var stackUpdateCmd = &cobra.Command{
 			utils.CheckErr(err)
 		}
 
+		if !utils.IsTerminal() && !output.CI {
+			fmt.Println("")
+			pterm.Warning.Println("Non-terminal detected, switching to non-interactive mode.")
```
Comment (jyecusch) [inline cmd/stack.go]:
```suggestion
			pterm.Warning.Println("non-interactive environment detected, switching to non-interactive mode.")
```

File: cmd/stack.go [hunk 6]
```diff
@@ -169,6 +179,12 @@ nitric stack down -s aws -y`,
 			}
 		}
 
+		if !utils.IsTerminal() && !output.CI {
+			fmt.Println("")
+			pterm.Warning.Println("Non-terminal detected, switching to non-interactive mode.")
```
Comment (jyecusch) [inline cmd/stack.go]:
```suggestion
			pterm.Warning.Println("non-interactive environment detected, switching to non-interactive mode.")
```

File: cmd/start.go [hunk 7]
```diff
@@ -41,6 +43,12 @@ var startCmd = &cobra.Command{
 		log.SetOutput(output.NewPtermWriter(pterm.Debug))
 		log.SetFlags(0)
 
+		if !utils.IsTerminal() && !output.CI {
+			fmt.Println("")
+			pterm.Warning.Println("Non-terminal detected, switching to non-interactive mode")
```
Comment (jyecusch) [inline cmd/start.go]:
```suggestion
			pterm.Warning.Println("non-interactive environment detected, switching to non-interactive mode.")
```

**Review by jyecusch — DISMISSED** [review]

**Review by HomelessDinosaur — DISMISSED** [review]

**Review by jyecusch — DISMISSED** [review]

**Review by jyecusch — APPROVED** [review]

**Review by HomelessDinosaur — APPROVED** [review]

**Pull request merged**